tools/flang2/flang2exe/llutil.cpp

```diff
@@ -2194,6 +2194,96 @@ metadata_args_need_struct(void)
   return ll_feature_metadata_args_struct(&llvm_get_current_module()->ir);
 }
 
+/**
+ * This function returns true for the types supported
+ * in function make_param_op
+ */
+bool
+should_preserve_param(const DTYPE dtype)
+{
+  switch (DTY(dtype)) {
+  // handled cases
+  case TY_ARRAY:
+    {
+      ADSC *ad = AD_DPTR(dtype);
+      SPTR size_sptr = AD_NUMELM(ad);
+      ISZ_T size = ad_val_of(size_sptr);
+      /* dont preserve zero-sized array, which would be optimized out later */
```

Comment:
Is this true even when compiling at `-O0`?

Reply:
BTW, please spell the first word with an apostrophe, or write "Do not".

Reply:
Yes that is optimized out in front-end itself. I shall update your comment. Thanks.